integration/index.2.test.js

```diff
@@ -148,6 +203,44 @@ describe('test/index.v2.test.js', () => {
     assert.strictEqual(require(path.join(cwd, 'node_modules', 'esbuild/package.json')).version, '0.15.14');
   });
 
+  it('deamon should working', async () => {
+    cwd = path.join(__dirname, './fixtures/esbuild');
+    await coffee
+      .fork(rapid, [
+        'install',
+        '--ignore-scripts',
+      ], {
+        cwd,
+      })
+      .debug()
+      .expect('code', 0)
+      .end();
+
+    const dirs = await fs.readdir(path.join(cwd, 'node_modules'));
+    assert.strictEqual(dirs.filter(dir => dir.includes('esbuild')).length, 2);
+    await assert.doesNotReject(fs.stat(path.join(cwd, 'node_modules/esbuild')));
+    assert.strictEqual(require(path.join(cwd, 'node_modules', 'esbuild/package.json')).version, '0.15.14');
+
+
+    const pidsInstance = new Pids(dirs);
+    let pids = await pidsInstance.getPids();
+    assert(pids.length > 0);
+    pids.forEach(pid => {
+      try {
+        execSync(`kill -9 ${pid}`);
+      } catch (err) {
+        console.error(`Failed to kill process with PID ${pid}. Error: ${err.message}`);
+      }
+    });
+    await new Promise(resolve => {
```

Comment:
Additionally, using 'sleep' in tests is not a good practice as it can lead to instability in the tests. It's preferable to use event-based approaches, where events can be triggered upon unexpected process exits or restarts.